docs/source/features/caching.md

```diff
@@ -0,0 +1,131 @@
+---
+title: Caching
+description: Automatically set HTTP cache headers! Save full responses in a cache!
+---
+
+Production apps often rely on caching for scalability.
+
+A single GraphQL request consists of running many different resolvers, each of which can have different caching semantics. Some fields may be uncacheable. Some fields may be cacheable for a few seconds, and others for a few hours. Some fields may have values that are the same for all users of your app, and other fields may vary based on the current session.
+
+Apollo Server provides a mechanism for server authors to declare fine-grained cache control parameters on individual GraphQL types and fields, both statically inside your schema using the `@cacheControl` directive and dynamically within your resolvers using the `info.cacheControl.setCacheHint` API.
+
+For each request, Apollo Server combines all the cache hints from all the queried fields and uses it to power several caching features. These features include **HTTP caching headers** for CDNs and browsers, and a GraphQL **full response cache**.
+
+
+## Defining cache hints
+
+You can define cache hints *statically* in your schema and *dynamically* in your resolvers.
+
+### Adding cache hints statically in your schema
+
+The easiest way to add cache hints is directly in your schema using the `@cacheControl` directive. Apollo Server automatically adds the definition of the `@cacheControl` directive to your schema when you create a new `ApolloServer` object with `typeDefs` and `resolvers`.
+
+You can apply `@cacheControl` to an individual field or to a type. Hints on a type apply to all fields that *return* objects of that type (not to the fields inside that type). Hints on fields override hints specified on the target type.  `@cacheControl` can specify `maxAge` (in seconds, like in an HTTP `Cache-Control` header) and `scope`, which can be `PUBLIC` (the default) or `PRIVATE`.
```

Comment:
I know what you mean by "Hints on a type apply to all fields that *return* objects of that type (not to the fields inside that type)", but I think this may require some clarification for it to make sense to people.

Reply:
OK, I added a bunch of specific examples.